### PR TITLE
snapcast:  0.27.0 -> 0.28.0; removed patch

### LINKED ISSUE #305522: Update request: snapcast 0.27.0 → 0.28.0
- Package name: snapcast
- Latest released version: 0.28.0

- Current version on the unstable channel: 0.27.0
- Current version on the stable/release channel: 0.27.0

- [x] Checked the [nixpkgs pull requests](https://github.com/NixOS/nixpkgs/pulls)

**Notify maintainers**

@fpletz

-----

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by kashw2**
## Description of changes

closes https://github.com/NixOS/nixpkgs/issues/305522

https://github.com/badaix/snapcast/releases/tag/v0.28.0

removed a patch that has been implemented upstream in this release

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by kashw2** [comment]
Result of `nixpkgs-review pr 305661` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>snapcast</li>
  </ul>
</details>

**Comment by kirillrdy** [comment]
tests failed in CI, but pass locally, going to kick it off again

@GrahamcOfBorg build snapcast.tests

**Comment by Scrumplex** [comment]
Result of `nixpkgs-review pr 305661` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>snapcast</li>
  </ul>
</details>

**Review by Scrumplex — APPROVED** [review]

File: pkgs/applications/audio/snapcast/default.nix [hunk 1]
```diff
@@ -9,24 +9,15 @@ assert pulseaudioSupport -> libpulseaudio != null;
...
     owner  = "badaix";
     repo   = "snapcast";
     rev    = "v${version}";
-    sha256 = "sha256-dlK1xQQqst4VQjioC7MZzqXwMC+JfqtvnD5lrOqGhYI=";
+    sha256 = "sha256-XxpiLojs1TI3qM7yXS+OPcHKo6pLcfJ+Eop7GjQ4HQw=";
   };
 
-  patches = [
-    # Can be removed with next release after 0.27.0
-    (fetchpatch {
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/snapcast/default.nix:23]: Please remove the unused fetchpatch from inputs.

File: pkgs/applications/audio/snapcast/default.nix [hunk 2]
```diff
@@ -9,24 +9,15 @@ assert pulseaudioSupport -> libpulseaudio != null;
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/snapcast/default.nix:9]: The assert can also be dropped. libpulseaudio is never null unless it is explicitly overwriten to that and then people probably know what they are doing.

Comment (kashw2) [reply pkgs/applications/audio/snapcast/default.nix]: fixed in https://github.com/NixOS/nixpkgs/pull/305661/commits/5c4361ae2fc982934e13dd30983fdc7764e0a1ca

File: pkgs/applications/audio/snapcast/default.nix (hunk 1 above)
Comment (kashw2) [reply pkgs/applications/audio/snapcast/default.nix:23]: fixed in https://github.com/NixOS/nixpkgs/pull/305661/commits/5c4361ae2fc982934e13dd30983fdc7764e0a1ca

**Review by fpletz — APPROVED** [review]

**Pull request merged**